### ISSUE TITLE
Resolve Duplicated Members in @azure/communication-react

### ISSUE THREAD

**Issue by scbedd** [Azure/communication-ui-library#763]
Given that the library actually exists here, this issue is re-filed from where I [originally filed it.](https://github.com/Azure/azure-sdk-for-js/issues/17429) 

[Visible Here](https://docs.microsoft.com/en-us/javascript/api/@azure/communication-react/?view=azure-node-latest)

![image](https://user-images.githubusercontent.com/45376673/131894466-a8372f7b-a58b-41c5-864e-47dddc023969.png)

Swapping the CI configuration to:

```
    {
      "name": "@azure/communication-react",
      "folder": "./dist/dist-esm/communication-react/src/"
    }
```

Actually provides the proper output. However, according to the package owners, that's actually the wrong place to go. 

The correct place should be...

```
    {
      "name": "@azure/communication-react",
      "folder": "./dist/"
    },
```

...according to the package owners. However, that is still generating the duplicates [as seen here.](https://review.docs.microsoft.com/en-us/javascript/api/@azure/communication-react/?view=azure-node-latest&branch=resolve-communication-react)

Need to adjust the structure of the package so that it can be properly documented. To determine an erroneous state, please run the following on the package.

How can you tell if there is a duplication with your configuration? 

```
npm install -g typedoc@0.15.0 type2docfx@beta node2docfx
npm install @azure/communication-react --prefix .
<within node_modules folder/@azure/communication-react, cd to the `folder` argument that we would provide to docs ci>
<so for the folder target of `/dist/`....>
cd node_modules/@azure/communication-react/dist
typedoc --json api.json --module commonjs --includeDeclarations --ignoreCompilerErrors --excludeExternals --stripInternal
```

Check `api.json`, and if you see the same member name twice, that's your repro.

The `api.json` is pretty hard to parse though. I have a `docker` image that can run the docs CI on a target package with a single command. That's probably the easiest way to repro. Will share offline.

**Comment by JamesBurnside** [comment]
I can confirm the duplicates are appearing because the typedoc is picking up definitions in the rolled up type file: `communication-react.d.ts`, as well as inside their original definition file inside the individual packlet directories:
| AreEqual defn 1 | AreEqual defn 2 |
| -- | -- |
| ![image](https://user-images.githubusercontent.com/2684369/133171463-172b6c83-cd00-4d3b-be64-3d1c1f6a81ac.png) | ![image](https://user-images.githubusercontent.com/2684369/133171471-0e700733-3af2-4a98-ae5e-d4f970d34294.png) |

I can also confirm that classes that are internal only and should not be displayed on the public api are being pulled from their original generated d.ts file:
![image](https://user-images.githubusercontent.com/2684369/133171767-54496488-9845-414a-b884-9aef53b5c303.png)

@scbedd is there an option to _only_ use the `communication-react.d.ts` file in the doc generation?

Otherwise @prprabhu-ms @PorterNan @DominikMe do you guys know if we need to include the original `.d.ts` files (e.g. `<package root>dist/dist-esm/acs-ui-common/src/areEqual.d.ts`) in the npm package? (i.e. is the rolled up `<package root>/communication-react.d.ts` sufficient?)

**Comment by scbedd** [comment]
@JamesBurnside you made some good recommendations here. I'll see if there is anything undocumented that this could work.

**Comment by ddematheu2** [comment]
@scbedd any update from your side on this? Looking to get an ETA for us to finalize reference docs. Thanks!

**Comment by scbedd** [comment]
Hey folks, got some spare time this evening to look through this again.

1. There is no option in the docs CI build to add typedoc `--exclude` options at this time. I could probably PR something like that, but would need to justify why we can't simply include a single `d.ts` file.
2. Adding a `typedoc.js` may actually work. **[1]** is how they call `typedoc`, so if it's in the package folder, it should get used. At least [that's what the documentation says](https://typedoc.org/guides/options/#options-1).
   1. My problem is locally I can't get this option to work. **[2]** is what I tried
3. I can just go after `dist-esm` short term until we either get an `exclude` function merged or adjust the structure of the package. I know this will eliminate the duplication, but what will be _messing up_ if we do it that way?

@JamesBurnside @ddematheu2 can we get on a call and just mess around with this thing realtime? I'll send a meeting invite.

**[1]**
![image](https://user-images.githubusercontent.com/45376673/134753053-3483129d-2c49-4cb7-b9ad-0b573e13d629.png)

**[2]**

Tried both `typedoc.js` and `typedoc.json`. `typedoc.js` crashes the process immediately, `typedoc.json` continues, but doesn't look  like it honors the `exclude` that I set.

Tried
```
{
    "exclude": ["dist-esm"]
}
```

and 

```
{
    "exclude": "dist-esm"
}
```

**Comment by JamesBurnside** [comment]
Thanks for looking into this! I'll confirm with folks on my side if we can remove the other `.d.ts` files and just include the rolled up one.

I'll try also playing around with the tsdoc command some more - maybe this feature only works in newer versions of the typedoc npm package.

For option 3 (going after just the `dist-esm` folder) - the reason we can't do that is it includes exports we don't export from the actual npm package. We have `@internal` only exports from the sub folders that should be kept out of the public API. We are in the process of marking these as `@internal` so it perhaps they can be filtered out as an alternative option but I think options 1 and 2 are better routes.

**Comment by JamesBurnside** [comment]
Confirmed two things:
* We should be able to use the `typedoc.js` in the following format: https://github.com/Azure/communication-ui-library/pull/833
  * Needs confirmed the tool DCOM uses can pick this file up
* If this doesn't work we do not need to include the `.d.ts` file (confirmed it does not impact build, treeshaking or type checking.

**Comment by scbedd** [comment]
We should be able to confirm the inclusion of the `typedoc.js` manually later today. If that fix gets published to a public dev feed we will be able to test it directly!

**Comment by JamesBurnside** [comment]
Oh great, published to our npm dev feed: (`@azure/communication-react@dev`) https://www.npmjs.com/package/@azure/communication-react/v/1.0.0-alpha-20210927.2200.0

**Comment by scbedd** [comment]
It's definitely targeted at `dist` right now. Here's the [commit to adjust targeting.](https://github.com/MicrosoftDocs/azure-docs-sdk-node/commit/6ca97f117bb920e99038e32f8b0fe809978ee314)

**Comment by DominikMe** [comment]
Late to the party.
> Otherwise @prprabhu-ms @PorterNan @DominikMe do you guys know if we need to include the original .d.ts files (e.g. <package root>dist/dist-esm/acs-ui-common/src/areEqual.d.ts) in the npm package? (i.e. is the rolled up <package root>/communication-react.d.ts sufficient?)

I would assume we only want one .d.ts file in the bundled package @JamesBurnside. Best to toy around with it in VS Code and see how intellisense deals with it. If there are multiple .d.ts, TS intellisense will probably to declaration merging, so you won't even recognize duplicates.

**Comment by prprabhu-ms** [comment]
We're only marking the symbols *in the package-public API* with @internal if it isn't intended to be public. So, if we're currently scraping symbols exported from modules deep in the packlets, that won't be fixed.

**Comment by JamesBurnside** [comment]
> We're only marking the symbols _in the package-public API_ with @internal if it isn't intended to be public. So, if we're currently scraping symbols exported from modules deep in the packlets, that won't be fixed.

Aren't we marking every exported function as either `@public`, `@internal` or `@private`?

**Comment by JamesBurnside** [comment]
This is now resolved. DCOM api no longer has duplicates or internal-only interfaces: https://docs.microsoft.com/en-us/javascript/api/@azure/communication-react/?view=azure-node-latest
